### DISCUSSION TITLE
没更新到2.0的时候还可以升级后就报错

### DISCUSSION THREAD

**Discussion by WinHongLiu** [tw93/Pake#428] [Q&A] [answered]
To build the Windows app, you need to install Rust and VS Build Tools.
See more in https://tauri.app/v1/guides/getting-started/prerequisites#installing

You have not provided an app icon, use the default icon.(use --icon option to assign an icon)
ping www.npmjs.com failed!, is not in China!

up to date, audited 243 packages in 1s

78 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

> pake-cli@2.0.4 build
> npm run tauri build --release

> pake-cli@2.0.4 tauri
> tauri build

   Compiling proc-macro2 v1.0.56
   Compiling quote v1.0.26
   Compiling syn v1.0.109
   Compiling serde v1.0.160
   Compiling serde_derive v1.0.160
   Compiling windows_x86_64_msvc v0.42.2
   Compiling log v0.4.17
   Compiling memchr v2.5.0
   Compiling getrandom v0.1.16
   Compiling winapi v0.3.9
   Compiling phf_shared v0.8.0
   Compiling getrandom v0.2.9
error: linker `link.exe` not found
  |
  = note: program not found

note: the msvc targets depend on the msvc linker but `link.exe` was not found

note: please ensure that Visual Studio 2017 or later, or Build Tools for Visual Studio were installed with the Visual C++ option.

note: VS Code is a different product, and is not sufficient.

error: could not compile `quote` due to previous error
warning: build failed, waiting for other jobs to finish...
error: could not compile `windows_x86_64_msvc` due to previous error
error: could not compile `log` due to previous error
error: could not compile `proc-macro2` due to previous error
error: could not compile `serde_derive` due to previous error
error: could not compile `serde` due to previous error
error: could not compile `syn` due to previous error
error: could not compile `memchr` due to previous error
error: could not compile `getrandom` due to previous error
error: could not compile `winapi` due to previous error
       Error failed to build app: failed to build app
file:///C:/Users/Administrator/AppData/Roaming/nvm/v16.20.0/node_modules/pake-cli/dist/cli.js:2021
                reject(new Error(`${code}`));
                       ^

Error: 1
    at file:///C:/Users/Administrator/AppData/Roaming/nvm/v16.20.0/node_modules/pake-cli/dist/cli.js:2021:24
    at C:\Users\Administrator\AppData\Roaming\nvm\v16.20.0\node_modules\pake-cli\node_modules\shelljs\src\exec.js:145:9
    at ChildProcess.exithandler (node:child_process:410:5)
    at ChildProcess.emit (node:events:513:28)
    at maybeClose (node:internal/child_process:1100:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:304:5)

**Comment by WinHongLiu** [comment]
node >=16.18.1   rust == 1.69.0

**Reply by Tlntin** [reply]
看起来是你的链接器出问题了，可能你要用vs2022安装工具检查一下你的环境了。

**Reply by WinHongLiu** [reply]
Downloading crates ...
warning: spurious network error (2 tries remaining): [92] Stream error in the HTTP/2 framing layer (HTTP/2 stream 0 was not closed cleanly: PROTOCOL_ERROR (err 1))
warning: spurious network error (2 tries remaining): [18] Transferred a partial file (transfer closed with 1123969 bytes remaining to read)
warning: spurious network error (2 tries remaining): [92] Stream error in the HTTP/2 framing layer (HTTP/2 stream 0 was not closed cleanly: PROTOCOL_ERROR (err 1))
warning: spurious network error (2 tries remaining): [92] Stream error in the HTTP/2 framing layer (HTTP/2 stream 0 was not closed cleanly: PROTOCOL_ERROR (err 1))
warning: spurious network error (2 tries remaining): [18] Transferred a partial file (transfer closed with 5842487 bytes remaining to read)
warning: spurious network error (2 tries remaining): [18] Transferred a partial file (transfer closed with 12986667 bytes remaining to read)
warning: spurious network error (2 tries remaining): [92] Stream error in the HTTP/2 framing layer (HTTP/2 stream 0 was not closed cleanly: PROTOCOL_ERROR (err 1))
warning: spurious network error (2 tries remaining): [92] Stream error in the HTTP/2 framing layer (HTTP/2 stream 0 was not closed cleanly: PROTOCOL_ERROR (err 1))
warning: spurious network error (2 tries remaining): [18] Transferred a partial file (transfer closed with 582035 bytes remaining to read)
warning: spurious network error (1 tries remaining): [92] Stream error in the HTTP/2 framing layer (HTTP/2 stream 0 was not closed cleanly: PROTOCOL_ERROR (err 1))
warning: spurious network error (1 tries remaining): [18] Transferred a partial file (transfer closed with 709785 bytes remaining to read)
warning: spurious network error (1 tries remaining): [18] Transferred a partial file (transfer closed with 6966684 bytes remaining to read)
warning: spurious network error (1 tries remaining): [92] Stream error in the HTTP/2 framing layer (HTTP/2 stream 0 was not closed cleanly: PROTOCOL_ERROR (err 1))
warning: spurious network error (1 tries remaining): [92] Stream error in the HTTP/2 framing layer (HTTP/2 stream 0 was not closed cleanly: PROTOCOL_ERROR (err 1))
error: failed to download from `https://crates.io/api/v1/crates/tao/0.15.8/download`

Caused by:
  [92] Stream error in the HTTP/2 framing layer (HTTP/2 stream 0 was not closed cleanly: PROTOCOL_ERROR (err 1))
       Error failed to build app: failed to build app
file:///C:/Users/Administrator/AppData/Roaming/nvm/v16.18.1/node_modules/pake-cli/dist/cli.js:2021
                reject(new Error(`${code}`));
                       ^

Error: 1
    at file:///C:/Users/Administrator/AppData/Roaming/nvm/v16.18.1/node_modules/pake-cli/dist/cli.js:2021:24
    at C:\Users\Administrator\AppData\Roaming\nvm\v16.18.1\node_modules\pake-cli\node_modules\shelljs\src\exec.js:145:9
    at ChildProcess.exithandler (node:child_process:410:5)
    at ChildProcess.emit (node:events:513:28)
    at maybeClose (node:internal/child_process:1100:16)
    at Socket.<anonymous> (node:internal/child_process:458:11)
    at Socket.emit (node:events:513:28)
    at Pipe.<anonymous> (node:net:301:12)
重装了三四次了,现在没法下crates 一直报错.墙内墙外都不行.大佬请问是什么问题.https://crates.io/api/v1/crates/tao/0.15.8/download   我下载和访问也没问题

**Reply by Tlntin** [reply]
换个nodejs版本试试（比如18，不过可能用处不大，感觉是你的网络出问题了。）

**Reply by WinHongLiu** [reply, marked as answer]
好的,谢谢大佬帮忙我用这个思路再排查一下
